src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -146,7 +146,16 @@ private function parseDefinitions(\DOMDocument $xml, $file, $defaults)
         foreach ($services as $service) {
             if (null !== $definition = $this->parseDefinition($service, $file, $defaults)) {
                 if ('prototype' === $service->tagName) {
-                    $this->registerClasses($definition, (string) $service->getAttribute('namespace'), (string) $service->getAttribute('resource'), (string) $service->getAttribute('exclude'));
+                    $excludes = array_map(function ($node) { return $node->nodeValue; }, $this->getChildren($service, 'exclude'));
+                    if ($service->hasAttribute('exclude')) {
```

Comment:
I would allow only tags XOR attribute (like done in similar cases in the loader AFAIK

Reply:
So in cas both are present, should I throw an exception?

Reply:
I would say yes